docs/helm-release.md

```diff
@@ -0,0 +1,95 @@
+
+## With `helm_release` Terraform Resource
+
+The [helm_release](https://registry.terraform.io/providers/hashicorp/helm/latest/docs/resources/release) resource  is the most basic way to provision a helm chart via terraform.
+
+Use this resource, if you need to control the lifecycle add-ons down to level of each add-on resource.
+
+## With `helm_releases` Variable
+
+You can use the `helm_releases` variable in [EKS Blueprints Add-ons](https://registry.terraform.io/modules/aws-ia/eks-blueprints-addons/aws/latest?tab=inputs) to provide a map of add-ons and their respective helm configuration. Under the hood, we just iterate through the provided map and pass it to the terraform [helm_release](https://registry.terraform.io/providers/hashicorp/helm/latest/docs/resources/release) resource.
+
+E.g.
+
+```hcl
+module "addons" {
+  source  = "aws-ia/eks-blueprints-addon/aws"
+  version = "1.1.0"
+
+  cluster_name      = "<cluster_name>"
+  cluster_endpoint  = "<cluster_endpoint>"
+  cluster_version   = "<cluster_version>"
+  oidc_provider_arn = "<oidc_provider_arn>"
+
+  # EKS add-ons
+  eks_addons = {
+    coredns = {}
+    vpc-cni = {}
+    kube-proxy = {}
+  }
+
+  # Blueprints add-ons
+  enable_aws_efs_csi_driver                    = true
+  enable_aws_cloudwatch_metrics                = true
+  enable_cert_manager                          = true
+  ...
+
+  # Pass in any number of Helm charts to be created for those that are not natively supported
+  helm_releases = {
+    prometheus-adapter = {
+      description      = "A Helm chart for k8s prometheus adapter"
+      namespace        = "prometheus-adapter"
+      create_namespace = true
+      chart            = "prometheus-adapter"
+      chart_version    = "4.2.0"
+      repository       = "https://prometheus-community.github.io/helm-charts"
+      values = [
+        <<-EOT
+          replicas: 2
+          podDisruptionBudget:
+            enabled: true
+        EOT
+      ]
+    }
+    gpu-operator = {
+      description      = "A Helm chart for NVIDIA GPU operator"
+      namespace        = "gpu-operator"
+      create_namespace = true
+      chart            = "gpu-operator"
+      chart_version    = "v23.3.2"
+      repository       = "https://nvidia.github.io/gpu-operator"
+      values = [
+        <<-EOT
+          operator:
+            defaultRuntime: containerd
+        EOT
+      ]
+    }
+  }
+
+  tags = local.tags
+}
+```
+
+With this pattern, the lifecycle of all your add-ons is tied to that of the `addons` module. This allows you to easily target the addon module in your terraform apply and destroy commands. E.g.
+
+```sh
+terraform apply -target=module.addons
+
+terraform destroy -target=module.addons
+```
+
+## With EKS Blueprints Add-on Module
```

Comment:
I struggle a ton on this one, and doesn't need to be solved here, but we should aim to be consistent - either `addon`/`addons` or `add-on`/`add-ons`

Reply:
Searching docs for other projects they use "Add-on" not "addon" when writing in sentences, titles, etc.. (ie Human readable)